### PR TITLE
feat(sink): support pause and resume for sink log store

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support `pause` and `resume` for sink log store. For the in mem log store, it will be no-op because it is not decoupled. For the decoupled kv log store, in the `init` method, we will pass the `pause_on_bootstrap` flag of barrier to the log writer. The log writer will pause the stream and the log reader will not read any data until the log writer resumes the stream.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/sink/log_store.rs [hunk 1]
```diff
@@ -304,6 +326,27 @@ where
     }
 }
 
+pub struct PausableLogWriter<W> {
+    inner: W,
+    is_paused: watch::Sender<bool>,
+}
```
Comment (tabVersion) [inline src/connector/src/sink/log_store.rs:332]: a little curious about where the struct used

File: src/stream/src/common/log_store_impl/kv_log_store/reader.rs [hunk 2]
```diff
@@ -149,6 +154,12 @@ impl<S: StateStore> LogReader for KvLogStoreReader<S> {
     }
 
     async fn next_item(&mut self) -> LogStoreResult<(u64, LogStoreReadItem)> {
+        while *self.is_paused.borrow_and_update() {
+            self.is_paused
+                .changed()
+                .await
+                .map_err(|_| anyhow!("unable to subscribe resume"))?;
+        }
```
Comment (tabVersion) [inline src/stream/src/common/log_store_impl/kv_log_store/reader.rs:162]: just confirmation, if it receives two resume mutations consecutively, the watcher's state won't change, right?

**Review by tabVersion — APPROVED** [review]
lgtm on my side, better add a test

File: src/connector/src/sink/log_store.rs (hunk 1 above)
Comment (hzxa21) [reply src/connector/src/sink/log_store.rs:332]: Same here. I didn't see this struct being used?

**Review by hzxa21 — APPROVED** [review]

File: src/stream/src/common/log_store_impl/kv_log_store/reader.rs (hunk 2 above)
Comment (wenym1) [reply src/stream/src/common/log_store_impl/kv_log_store/reader.rs:162]: Yes. The watch channel only retain the last sent value. 

https://docs.rs/tokio/latest/tokio/sync/watch/index.html

**Pull request merged**